packages/features/sales-escrows/modals/ConfirmEndSaleModal.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState } from 'react';
+import Slide from '@mui/material/Slide';
+import { TransitionProps } from '@mui/material/transitions';
+import { AuthContext } from '@dapp/features-authentication';
+import { LoadingContainer } from '@dapp/features-components';
+import { useSnackbar } from 'notistack';
+import { Container, Flex, Modal, Button, HR } from '@origyn-sa/origyn-art-ui';
+import { useDebug } from '@dapp/features-debug-provider';
+
+const Transition = React.forwardRef(
+  (
+    props: TransitionProps & {
+      children: React.ReactElement<any, any>;
+    },
+    ref: React.Ref<unknown>,
+  ) => <Slide direction="up" ref={ref} {...props} />,
+);
+
+Transition.displayName = 'Transition';
+
+interface ConfirmEndSaleModalProps {
+  onModalOpen: boolean;
+  onModalClose: () => void;
+  currentToken: string;
+  onSaleCancelled?: () => void;
+  onProcessing?: (boolean) => void;
+}
+
+export const ConfirmEndSaleModal = ({
+  onModalOpen,
+  onModalClose,
+  currentToken,
+  onSaleCancelled,
+  onProcessing,
+}: ConfirmEndSaleModalProps) => {
+  const { actor } = React.useContext(AuthContext);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const { enqueueSnackbar } = useSnackbar() || {};
+  const [confirmed, setConfirmed] = useState(false);
+  const debug = useDebug();
+
+  const onEndSaleConfirm = async (confirm = false) => {
+    if (!confirm) {
+      onModalClose();
+    } else if (isLoading || !actor) {
+      return;
+    }
+    try {
+      setIsLoading(true);
+      setConfirmed(true);
+      onProcessing?.(true);
+
+      const endSaleResponse = await actor.sale_nft_origyn({
+        end_sale: currentToken,
+      });
+      if (endSaleResponse.ok) {
+        enqueueSnackbar(`You have successfully ended the sale for ${currentToken}.`, {
+          variant: 'success',
+          anchorOrigin: {
+            vertical: 'top',
+            horizontal: 'right',
+          },
+        });
+        onSaleCancelled();
+      }
+      enqueueSnackbar(`Error: ${endSaleResponse.err.flag_point}.`, {
```

Comment:
This immediately follows the success snackbar message. There is no conditional, so won't it appear every time?